### PR TITLE
Add `system.device` to scripting

### PR DESCRIPTION
**Pull request by jsundra**
The scripting api itself isn't bad- but getting `IDeviceMetaProvider` into `SystemJsApi` was kind of gross. I'm thinking in the future, we might want to have some interface lookup/resolver that is responsible for managing instances rather than `ScriptCollectionRunner` crudely creating a single instance on first run :/

Open to ideas of how to make this better, was trying to balance time/profit.

### CODE REVIEW

File: Assets/Source/Player/IUX/Content/ScriptCollectionRunner.cs [hunk 1]
```diff
@@ -101,6 +108,11 @@ private enum SetupState
             _elementJsFactory = elementJsFactory;
             _root = root;
             _element = element;
+
+            if (_systemJsApi == null)
+            {
+                _systemJsApi = new SystemJsApi(deviceMetaProvider);
+            }
```
Comment (thegoldenmule) [inline Assets/Source/Player/IUX/Content/ScriptCollectionRunner.cs]: Haaaah! This seems particularly evil. I would suggest just using a static reference to `SystemJsApi`.

**Comment by jsundra** [comment]
Pretty sure there's only just this 1 instance since I removed its self-creating instance. But where would you suggest constructing the static reference from? The other thought I had was to leave it managing its own static instance still and referencing that- but then calling something like `SystemJsApi.SetDeviceMetaProvider(x)`, but that also seemed gross :/

**Comment by thegoldenmule** [comment]
What about something like this inside `EnkluPlayerModule`:

```
SystemJsApi.Instance.DeviceMetaProvider = binder.GetInstance<IDeviceMetaProvider>();
```

It's a tad ugly, but setting up these big objects inside the IoC container is fairly contained. Then in `UnityScriptingHost`:

```
SetValue('system', SystemJsApi.Instance);
```

**Comment by jsundra** [comment]
I like it! Seems like a pretty good way to keep the configuration high level.

File: Assets/Source/Player/Scripting/Interface/App/SystemJsApi.cs [hunk 2]
```diff
@@ -8,7 +10,26 @@ public class SystemJsApi
...
         /// The instance.
         /// </summary>
-        public static SystemJsApi Instance = new SystemJsApi();
+        public static SystemJsApi Instance = new SystemJsApi();
+
+        /// <summary>
+        /// Provider for getting DeviceMeta.
+        /// </summary>
+        public static IDeviceMetaProvider DeviceMetaProvider {
+            set {
```
Comment (thegoldenmule) [inline Assets/Source/Player/Scripting/Interface/App/SystemJsApi.cs]: Braces on separate lines.

**Comment by thegoldenmule** [comment]
COOL

**Pull request merged**